src/rpc/blockchain.cpp

```diff
@@ -1689,28 +1710,59 @@ UniValue getblockchaininfo(const JSONRPCRequest &request) {
     //     BIP9SoftForkDescPushBack(bip9_softforks, consensusParams, static_cast<Consensus::DeploymentPos>(pos));
     // }
     for (int i = 0; i < static_cast<int>(EUpdate::MAX_VERSION_BITS_DEPLOYMENTS); ++i) {
-        StateInfo state = Updates().State( static_cast<EUpdate>(i), ::ChainActive().Tip());
+        StateInfo state = Updates().State(static_cast<EUpdate>(i), ::ChainActive().Tip());
         if (state.State == EUpdateState::Unknown)
             continue;
-        const Update* update = Updates().GetUpdate(static_cast<EUpdate>(i));
+        const Update *update = Updates().GetUpdate(static_cast<EUpdate>(i));
         UniValue rv(UniValue::VOBJ);
         switch (state.State) {
-            case EUpdateState::Defined: rv.pushKV("status", "defined"); break;
-            case EUpdateState::Failed: rv.pushKV("status", "failed"); break;
-            case EUpdateState::LockedIn: rv.pushKV("status", "locked_in"); break;
-            case EUpdateState::Voting: rv.pushKV("status", "voting"); break;
-            case EUpdateState::Active: rv.pushKV("status", "active"); break;
+            case EUpdateState::Defined:
+                rv.pushKV("status", "defined");
+                break;
+            case EUpdateState::Failed:
+                rv.pushKV("status", "failed");
+                break;
+            case EUpdateState::LockedIn:
+                rv.pushKV("status", "locked_in");
+                break;
+            case EUpdateState::Voting:
+                rv.pushKV("status", "voting");
+                break;
+            case EUpdateState::Active:
+                rv.pushKV("status", "active");
+                break;
         }
         if (state.State == EUpdateState::Voting) {
             rv.pushKV("bit", update->Bit());
         }
         rv.pushKV("start_height", update->StartHeight());
         rv.pushKV("round_size", update->RoundSize());
         rv.pushKV("voting_period", update->VotingPeriod());
-        bip9_softforks.pushKV(update->Name(), rv);
+        if (state.voteStats.minerVoteResult) {
```

Comment:
FYI - this state object may be holding two pointers managed by the UpdateManager.  They could become invalid at any time.